app/code/Magento/Catalog/Test/Unit/Model/ProductRepositoryTest.php

```diff
@@ -345,6 +346,66 @@ function ($value) {
         $this->objectManager->setBackwardCompatibleProperty($this->model, 'mediaProcessor', $mediaProcessor);
     }
 
+    /**
+     * Test save product with store id 0
+     *
+     * @param array $productData
+     * @return void
+     * @dataProvider getProductData
+     */
+    public function testSaveForAllStoreViewScope(array $productData): void
+    {
+        $this->productFactory->method('create')->willReturn($this->product);
+        $this->product->method('getSku')->willReturn($productData['sku']);
+        $this->extensibleDataObjectConverter
+            ->expects($this->once())
+            ->method('toNestedArray')
+            ->willReturn($productData);
+        $this->resourceModel->method('getIdBySku')->willReturn(100);
+        $this->resourceModel->expects($this->once())->method('validate')->willReturn(true);
+        $this->product->expects($this->at(14))->method('setData')->with('store_id', $productData['store_id']);
+
+        $this->model->save($this->product);
+    }
+
+    /**
+     * Product data provider
+     *
+     * @return array
+     */
+    public function getProductData(): array
+    {
+        return [
+            [
+                [
+                    'sku' => 'sku',
+                    'name' => 'product',
```

Comment:
`const STUB_PRODUCT_NAME` to show that actual product name does not matter, it's just stub

Reply:
done